crates/ruff_python_formatter/src/expression/expr_tuple.rs

```diff
@@ -19,14 +19,17 @@ pub enum TupleParentheses {
     Default,
     /// Effectively `Some(Parentheses)` in `Option<Parentheses>`
     Expr(Parentheses),
+    /// TODO(cnpryer): Why not TupleParentheses::Expr(Parentheses::Never)?
+    Never,
 
-    /// Black omits parentheses for tuples inside of subscripts except if the tuple is parenthesized
-    /// in the source code.
-    Subscript,
-
-    /// Handle the special case where we remove parentheses even if they were initially present
+    /// Handle special cases where parentheses are not introduced if they are not already present.
+    ///
+    /// Black omits parentheses for tuples inside subscripts except if the tuple is already
+    /// parenthesized in the source code.
+    NeverIntroduce,
```

Comment:
I'd rename this to Preserve to match `Parentheses::Preserve`

Reply:
Could you add these as black formatting examples to the source code? I think it's easier for someone later to follow what is actually happening if you can see the cases.
```python
x[a, :]
x[a, b:]
x[(a, b):]
```

Reply:
Would you prefer I add these fixtures in a separate PR? I might do that to get started and if you want it here we can just close that, but my thought process is that if we decide to ditch any `TupleParentheses` changes we'd still want those fixtures.

Reply:
I think I misunderstood. You wanted these in the comments (which shouldn't be `///`, I'll fix that too).

Reply:
89268f0d6ed17c877f6d73ea2926f44fab153b20 (potentially resolved)